src/gradata/daemon.py

```
@@ -337,6 +383,192 @@ def _handle_end_session(self) -> None:
+        d = self.daemon
+        pending = 0
+        try:
+            with d._brain_lock:
+                lessons = d._brain._load_lessons()
+                pending = sum(1 for le in lessons
+                              if le.state in (LessonState.INSTINCT, LessonState.PATTERN))
+                d._brain.emit("CHECKPOINT", "plugin.pre_compact", {
+                    "session_id": session_id, "reason": reason, "pending_lessons": pending,
+                })
+        except Exception:
+            pass
```

Ruff (0.15.9)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)